tests/nlp/utils/test_properties.py

```diff
@@ -83,30 +83,185 @@ def test_calculate_average_sentence_length_property(tweet_emotion_train_test_tex
-    result_none_text = calculate_default_properties([None], include_properties=['Readability Score'])[0]
+    result = calculate_builtin_properties(test_text, include_properties=['Readability Score'])[0]
+    result_none_text = calculate_builtin_properties([None], include_properties=['Readability Score'])[0]
 
     # Assert
-    assert_that(result['Readability Score'][0: 10], equal_to([
-        102.045, 97.001, 80.306, 67.755, 77.103, 71.782, np.nan, 75.5, 70.102, 95.564
-    ]))
+    assert_that(result['Readability Score'][0: 10], equal_to([102.045, 97.001, 80.306, 67.755, 77.103,
+                                                            71.782, np.nan, 75.5, 70.102, 95.564]))
     assert_that(result_none_text['Readability Score'], equal_to([np.nan]))
 
 
+def test_calculate_count_unique_urls():
+
+    # Arrange
+    text_data = [
+        'Please contact me at abc.ex@example.com.',
+        'For more information, visit our website: https://deepchecks.com/.',
+        'Email us at info@example.com or visit our website http://www.example.com for assistance.',
+        'For any inquiries, send an email to support@example.com.',
+        'The results were found at http://www.google.com and it redirects to'
+        'https://www.deepchecks.com and there we can find the links to all social medias such'
+        'as http://gmail.com, https://fb.com, https://www.deepchecks.com, https://www.google.com'
+    ]
+
+    # Act
+    result = calculate_builtin_properties(text_data, include_properties=['Count Unique URLs'])[0]
+    result_none_text = calculate_builtin_properties([None], include_properties=['Count Unique URLs'])[0]
+
+    # Assert
+    assert_that(result['Count Unique URLs'], equal_to([0, 1, 1, 0, 5]))
+    assert_that(result_none_text['Count Unique URLs'], equal_to([0]))
+
+
+def test_calculate_count_urls():
+
+    # Arrange
+    text_data = [
+        'Please contact me at abc.ex@example.com.',
+        'For more information, visit our website: https://deepchecks.com/.',
+        'Email us at info@example.com or visit our website http://www.example.com for assistance.',
+        'For any inquiries, send an email to support@example.com.',
+        'The results were found at http://www.google.com and it redirects to'
+        'https://www.deepchecks.com and there we can find the links to all social medias such\
+        as http://gmail.com, https://fb.com, https://www.deepchecks.com, https://www.google.com'
+    ]
+
+    # Act
+    result = calculate_builtin_properties(text_data, include_properties=['Count URLs'])[0]
+    result_none_text = calculate_builtin_properties([None], include_properties=['Count URLs'])[0]
+
+    # Assert
+    assert_that(result['Count URLs'], equal_to([0, 1, 1, 0, 6]))
+    assert_that(result_none_text['Count URLs'], equal_to([0]))
+
+
+def test_calculate_count_unique_email_addresses():
+
+    # Arrange
+    text_data = [
```

Comment:
You can create this text data as a fixture at the beggining of this file.

Reply:
Done